File: library/src/test/java/me/proxer/library/enums/NotificationTypeTest.java
package me.proxer.library.enums;

import me.proxer.library.ProxerTest;
import org.junit.jupiter.api.Test;

import java.io.IOException;

import static org.assertj.core.api.Assertions.assertThat;

class NotificationTypeTest extends ProxerTest {

    @Test
    void testDefault() throws IOException {
        final NotificationType notificationType = api.moshi().adapter(NotificationType.class).fromJson("\"ticket\"");

        assertThat(notificationType).isSameAs(NotificationType.TICKET);
    }

    @Test
    void testFallback() throws IOException {
        final NotificationType notificationType = api.moshi().adapter(NotificationType.class).fromJson("\"xyz\"");

        assertThat(notificationType).isSameAs(NotificationType.OTHER);
    }
}

File: library/src/test/kotlin/me/proxer/library/enums/NotificationTypeTest.kt
package me.proxer.library.enums

import com.serjltt.moshi.adapters.FallbackEnum
import com.squareup.moshi.Moshi
import org.amshove.kluent.shouldBe
import org.junit.jupiter.api.Test

/**
 * @author Ruben Gees
 */
class NotificationTypeTest {

    private val adapter = Moshi.Builder()
        .add(FallbackEnum.ADAPTER_FACTORY)
        .build()
        .adapter(NotificationType::class.java)

    @Test
    fun testDefault() {
        adapter.fromJson("\"ticket\"") shouldBe NotificationType.TICKET
    }

    @Test
    fun testFallback() {
        adapter.fromJson("\"xyz\"") shouldBe NotificationType.OTHER
    }
}
